components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
-    real(r8) surfric(pcols)            ! surface friction velocity
-    real(r8) obklen(pcols)             ! Obukhov length
-    real(r8) :: fh2o(pcols)            ! h2o flux to balance source from methane chemistry
-    real(r8) :: tmp_q     (pcols,pver) ! tmp space
-    real(r8) :: tmp_cldliq(pcols,pver) ! tmp space
-    real(r8) :: tmp_cldice(pcols,pver) ! tmp space
-    real(r8) :: tmp_t     (pcols,pver) ! tmp space
-    real(r8) :: ftem      (pcols,pver) ! tmp space
-    real(r8), pointer, dimension(:) :: static_ener_ac_2d ! Vertically integrated static energy
-    real(r8), pointer, dimension(:) :: water_vap_ac_2d   ! Vertically integrated water vapor
-
-    ! physics buffer fields for total energy and mass adjustment
-    integer itim_old, ifld
-
-    real(r8), pointer, dimension(:,:) :: tini
-    real(r8), pointer, dimension(:,:) :: cld
-    real(r8), pointer, dimension(:,:) :: qini
-    real(r8), pointer, dimension(:,:) :: cldliqini
-    real(r8), pointer, dimension(:,:) :: cldiceini
-    real(r8), pointer, dimension(:,:) :: dtcore
-    real(r8), pointer, dimension(:,:) :: ast     ! relative humidity cloud fraction 
-    logical :: do_clubb_sgs 
-#ifdef MAML
-    real(r8) :: factor_xy    ! for converting from CRM to GCM-level
-    integer  :: ii           ! loop index for CRM
-#endif
-
-    !DCAPE-ULL: physics buffer fields to compute tendencies for dcape
-    real(r8), pointer, dimension(:,:) :: t_star   ! temperature
-    real(r8), pointer, dimension(:,:) :: q_star   ! moisture
-    logical :: do_zmconv_trigdcape_ull            ! switch if using dcape-ull as trigger for ZM convection
-                                                  ! default to false, to bbe made controllable by nml
-    ! Debug physics_state.
-    logical :: state_debug_checks
-
-    logical :: l_tracer_aero
-    logical :: l_vdiff
-    logical :: l_rayleigh
-    logical :: l_gw_drag
-    logical :: l_ac_energy_chk
-
-    !
-    !-----------------------------------------------------------------------
-    !
-    lchnk = state%lchnk
-    ncol  = state%ncol
-
-    nstep = get_nstep()
-    
-    call phys_getopts( do_clubb_sgs_out       = do_clubb_sgs, &
-                       state_debug_checks_out = state_debug_checks &
-                      ,l_tracer_aero_out      = l_tracer_aero      &
-                      ,l_vdiff_out            = l_vdiff            &
-                      ,l_rayleigh_out         = l_rayleigh         &
-                      ,l_gw_drag_out          = l_gw_drag          &
-                      ,l_ac_energy_chk_out    = l_ac_energy_chk    &
-                     )
-
-    ! Adjust the surface fluxes to reduce instabilities in near sfc layer
-    if (phys_do_flux_avg()) then 
-       call flux_avg_run(state, cam_in,  pbuf, nstep, ztodt)
-    endif
-
-    ! Validate the physics state.
-    if (state_debug_checks) &
-         call physics_state_check(state, name="before tphysac")
-
-    call t_startf('tphysac_init')
```

Comment:
It looks like we lost the `tphysac_init` timers. I'm not sure this is a good thing. That said, they ought to appear at the top of the routine.

Reply:
I removed several timers in physpkg because they seemed to be excessive. I remember combing through the detailed timing file and being overwhelmed with all these timers with very small values. I think it's better to strip most of this away and add them as we need them.

Reply:
added timers back in

Reply:
FYI - some of these timers are there because, when scanning the code, the question kept coming up "how expensive is that call?". "Documenting" that the cost is small eliminated a lot of experiments where a timer is added to check, and then removed (and the result forgotten or a  new person decides to look at this). We also do not have a convenient way of documenting the difference between the recorded time for the parent and the accumulated time for the children timers (though easy enough to do postprocessing). Adding timers to provide coverage for all code in a major routine can be useful in certain circumstances.

Reply:
Also, having timers (only) inside the routine being timed loses the call location information if that routines is called from more than one location. It also loses the call cost (which is hopefully small, but sometimes is not due to excessive array copying, for example).

Reply:
Thanks for insight @worleyph